proto/ibc/applications/fee/v1/query.proto

```diff
@@ -53,3 +59,20 @@ message QueryIncentivizedPacketResponse {
   // Incentivized_packet
   ibc.applications.fee.v1.IdentifiedPacketFee incentivized_packet = 1;
 }
+
+// QueryIncentivizedPacketsForChannelRequest is the request type for querying for all incentivized packets
+// for a specific channel
+message QueryIncentivizedPacketsForChannelRequest {
+  // pagination defines an optional pagination for the request.
+  cosmos.base.query.v1beta1.PageRequest pagination = 1;
+  string                                port_id    = 2;
+  string                                channel_id = 3;
+  // Height to query at
+  uint64 query_height = 4;
+}
+
+// QueryIncentivizedPacketsResponse is the response type for the incentivized packets RPC
+message QueryIncentivizedPacketsForChannelResponse {
+  // Map of all incentivized_packets
+  repeated ibc.applications.fee.v1.IdentifiedPacketFees incentivized_packets = 1;
```

Comment:
Should this be non-nullable?

Reply:
I thought so? But didn't we have conversation on the gRPC's needing to be pointers?

Reply:
Hmm, I do vaguely recall something. I can't remember why, or think of a reason it needs to a pointer type right now tho! Happy to leave as is, your call!